### PR TITLE
#98 - Fix native worker compatibility

### PR DESCRIPTION
**Pull request by Fordi**
Fixes: #98 

This PR modifies the node module initialization to give it minimal context-awareness, such that it can work within a `node:worker_threads` worker.

See: https://github.com/nodejs/node/blob/main/doc/api/addons.md#context-aware-addons

### CODE REVIEW

File: binding.gyp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "targets": [
         {
-            "target_name": "fs-ext", 
+            "target_name": "fs_ext",
```
Comment (Fordi) [inline binding.gyp:4]: Note: NODE_MODULE_INIT uses NODE_GYP_MODULE_NAME as an identifier, so this must be a valid identifier.

File: package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "node": ">= 8.0.0"
   },
   "dependencies": {
-    "nan": "^2.14.0"
+    "nan": "^2.21.0"
```
Comment (Fordi) [inline package.json:22]: Build was failing with nan 2.14.

**Comment by baudehlo** [comment]
I haven't - someone else took over maintenance of this module, but maybe
they are awol now. It looks fine, and I'm happy to merge.

**Review by baudehlo — APPROVED** [review]

**Comment by Fordi** [comment]
Been a week, the maintainer hasn't responded.  Please merge.

**Comment by baudehlo** [comment]
I merged it, but I've been trying to add Actions tests (it was originally setup using Travis), and the tests don't pass: https://github.com/baudehlo/node-fs-ext/pull/110

**Comment by Fordi** [comment]
Ah.  Run `git pull --rebase origin master` on your patch branch.  The `fs-ext.cc` in there still has `NODE_MODULE` [at the bottom](https://github.com/baudehlo/node-fs-ext/blob/baudehlo-patch-1/fs-ext.cc#L619), not `NODE_MODULE_INIT`.

**Pull request merged**